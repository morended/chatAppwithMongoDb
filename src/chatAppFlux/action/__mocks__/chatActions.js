var chatActionsMock = jest.genMockFromModule('./../chatActions.js');

module.exports = chatActionsMock;
