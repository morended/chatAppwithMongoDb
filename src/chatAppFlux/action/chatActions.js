import alt from "../alt.js"
import http from "superagent"

class chatActions {

  //registerUser(payload) {
    //return (dispatch) => {
      //http
        //.post('/api/register')
        //.set('Content-Type', 'application/json')
        //.query(payload)
        //.send(payload)
        //.end((err, res) => this.handleResponse(err, res, dispatch));
    //};
  //}

  //loginUser(payload) {
    //return (dispatch) => {
      //http
        //.post('/api/login')
        //.set('Content-Type', 'application/json')
        //.query(payload)
        //.send(payload)
        //.end((err, res) => this.handleResponse(err, res, dispatch));
    //};
  //}

  updateMessages(payload) {
    return (dispatch) => {
      http
        .post('/api/chats/sendMessage')
        .set('Content-Type', 'application/json')
        .query(payload)
        .send(payload)
        .end((err, res) => this.handleResponse(err, res, dispatch));
      };
  }

  handleResponse(error, result, dispatch) {
    if (error) {
      console.log('An error occurred:', error);
      alert('An error occurred: ', error);
      return;
    }

    dispatch({
      data: result.body
    });
  }
}

export default alt.createActions(chatActions);
