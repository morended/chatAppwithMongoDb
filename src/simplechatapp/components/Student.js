"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

class Student extends React.Component {
  constructor() {
    super()
    this.state = {
      message: ""
    }
    this._handleClick=this._handleClick.bind(this);
  this._handleOnChange=this._handleOnChange.bind(this);
  }
  _handleOnChange(event){

   this.setState({message:event.target.value})
 }
 _handleClick(){
   this.props.messages(this.props.id+this.state.message);
   this.state.message="";
 }
  render() {


return(
  <div>
{this.props.id}<input type="text"  value={this.state.message} onChange={this._handleOnChange}/>
</div>
<input type="Button" value="send"  onClick={this._handleClick}/>
  </div>
  )}

export default Student;