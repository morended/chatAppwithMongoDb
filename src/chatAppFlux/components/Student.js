"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import chatActions from '../action/chatActions'

class Student extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      message: ""
    };
  }

  _handleOnChange = (e) => {
     this.setState({
       message:e.target.value
     });
   }

   _handleClick = (e) => {
      e.preventDefault();
      chatActions.updateMessages({
        id: this.props.id,
        message: this.state.message
      });

      this.setState({
        message: ""
      });
   }

   render() {
     return (
        <div>
          Student {this.props.id}:
          <input
            type="text"
            value={this.state.message}
            onChange={this._handleOnChange}/>

          <button
            onClick={this._handleClick}>
            Send
          </button>
        </div>
     );
  }
}

export default Student;
