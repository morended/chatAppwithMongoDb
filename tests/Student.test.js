import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';

import chatActions from '../src/chatAppFlux/action/chatActions.js';
import Student from '../src/chatAppFlux/components/Student.js';

jest.mock('../src/chatAppFlux/action/chatActions.js');

describe('Student component (snapshot)', () => {

  it('renders correctly', () => {
    const component = renderer.create(
      <Student id='1'/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('Student component', () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <Student id='1'/>
    );
  });

  it('renders the id', () => {
    expect(component.find('div').text()).toContain("Student 1");
  });

  it('sets initial state', () => {
    expect(component.state().message).toEqual("");
  });

  it('renders text box', () => {
    expect(component.find('input').length).toEqual(1);
  });

  it('renders button', () => {
    let button = component.find('button');
    expect(button.length).toEqual(1);
    expect(button.text()).toEqual('Send');
  });

  it('updates state on text change', () => {
    let evnt = { target : {value: 'test'} };
    evnt.preventDefault = jest.fn();
    component.find('input').simulate('change', evnt);
    expect(component.state().message).toEqual(evnt.target.value);
  });

  it('resets state on button click', () => {
    let evnt = { target : { value: 'test'} };
    evnt.preventDefault = jest.fn();
    component.find('button').simulate('click', evnt);
    expect(component.state().message).toEqual('');
  });

  it('call chatActions updateMessages on button click', () => {
    let evnt = { target : { value: 'test'} };
    evnt.preventDefault = jest.fn();

    //clear mock information from previous tests
    chatActions.updateMessages.mockClear();

    component.setState({message: 'mystate'});
    component.find('button').simulate('click', evnt);
    expect(chatActions.updateMessages).toHaveBeenCalledWith({id:'1', message: 'mystate'})
  });

});
