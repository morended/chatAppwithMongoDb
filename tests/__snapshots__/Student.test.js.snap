// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student component (snapshot) renders correctly 1`] = `
<div>
  Student 
  1
  :
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Send
  </button>
</div>
`;
